op-challenger/fault/game.go

```diff
@@ -10,4 +19,65 @@ type Game interface {
 		claim  Claim
 		parent Claim
 	}
+
+	// Retrieve the parent of a given claim.
+	GetClaimParent(claim Claim) Claim
+}
+
+// gameState is a struct that implements the [Game] interface.
+type gameState struct {
+	// claims is a map of claim gindex to
+	claims map[uint64]Claim
+}
+
+// NewGameState returns a new game state.
+func NewGameState() *gameState {
+	return &gameState{
+		claims: make(map[uint64]Claim),
```

Comment:
There are allowed to be multiple claims at a specific gindex. The claim index + value is what must be unique (& claim index really means attacking/defending a specific node). I'd probably follow the contract representation a little more & have have claims relate to each other with custom metadata. It's fine to make a `node` struct & use it here.